crates/epoch-snark/src/api/prover.rs

```diff
@@ -18,13 +23,17 @@ pub fn prove(
     rng: &mut impl rand::Rng,
 ) -> Result<Groth16Proof<CPCurve>, SynthesisError> {
     let composite_hasher = CompositeHasher::new().unwrap();
+    let try_and_increment = TryAndIncrement::new(&composite_hasher);
+
     // Generate the CRH per epoch
     let mut message_bits = Vec::with_capacity(transitions.len());
     let mut epochs = Vec::with_capacity(transitions.len());
     for transition in transitions {
         let block = &transition.block;
-        let epoch_bytes = block.encode_to_bytes().unwrap();
-        let crh_bytes = composite_hasher.crh(&[], &epoch_bytes, 0).unwrap();
+        let epoch_bytes = block.encode_to_bytes(false).unwrap();
+        let (_, counter) = try_and_increment
```

Comment:
```suggestion
        // We need to find the counter so that the CRH hash we use will eventually result on an element on the curve
        let (_, counter) = try_and_increment
```